src/daft-parquet/src/read.rs

```diff
@@ -329,21 +319,6 @@ async fn read_parquet_single_into_arrow(
         }?;
     };
 
-    let expected_num_columns = if let Some(columns) = columns {
```

Comment:
Why are we deleting this verification logic?

Reply:
(see https://github.com/Eventual-Inc/Daft/pull/1990#discussion_r1520118516)